spring-rabbit/src/test/java/org/springframework/amqp/rabbit/annotation/EnableRabbitIntegrationTests.java

```diff
@@ -661,6 +686,35 @@ public void handleDifferent(Foo2 foo) {
 		public void handleWithAutoStartFalse(String foo) {
 		}
 
+		@RabbitListener(bindings = {
+				@QueueBinding(
+					value = @Queue(value = "auto.headers1", autoDelete = "true",
+									arguments = @Argument(name = "x-message-ttl", value = "10000",
+															type = "java.lang.Integer")),
+					exchange = @Exchange(value = "auto.headers", type = ExchangeTypes.HEADERS, autoDelete = "true"),
+					arguments = {
+							@Argument(name = "x-match", value = "all"),
+							@Argument(name = "foo", value = "bar"),
+							@Argument(name = "baz")
+					}
+				),
+				@QueueBinding(
+					value = @Queue(value = "auto.headers2", autoDelete = "true",
+									arguments = @Argument(name = "x-message-ttl", value = "10000",
+															type = "#{T(java.lang.Integer)}")),
+					exchange = @Exchange(value = "auto.headers", type = ExchangeTypes.HEADERS, autoDelete = "true"),
+					arguments = {
+							@Argument(name = "x-match", value = "any"),
+							@Argument(name = "foo", value = "bax"),
+							@Argument(name = "#{'baz'}", value = "#{'fiz'}")
+					}
+				)
+			}
+		)
+		public String handleWithHeadersExchange(String foo) {
```

Comment:
I think you will agree with me that this `@RabbitListener` declaration looks ugly. Like it would be better that we haven never introduced .
But the cat is out of the bag, plus there is some request from the community.
And yes: no one makes you to use this approach :smile: